.github/workflows/code_analysis.yml

```diff
@@ -18,18 +18,19 @@ jobs:
         (github.event_name == 'pull_request' && !contains(github.event.pull_request.title, '[skip-ci]'))
 
     runs-on: ubuntu-latest
-    env:
-      TRAVIS_BRANCH: ${{ github.base_ref }} 
-      TRAVIS_PULL_REQUEST_BRANCH: ${{ github.head_ref }}
-      BASE_COMMIT: ${{ github.event.pull_request.base.sha }}
     steps:
-    - uses: actions/checkout@v4
-    - name: Fetch base sha
-      run: git fetch --depth=1 origin +${{github.event.pull_request.base.sha}}:origin/base_sha
     - name: install clang-format
       run: sudo apt-get install -y clang-format
-    - name: run clang-format script
-      run: .ci/format_script.sh
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 1024
+        ref: ${{ github.event.pull_request.head.sha }}
+    - name: Fetch base sha
+      run: git fetch --depth=1024 origin +${{github.event.pull_request.base.sha}}:origin/base_sha
+    - name: Compute merge-base
+      run: echo "MERGE_BASE=$(git merge-base origin/base_sha HEAD)" >> "$GITHUB_ENV"
+    - name: run clang-format
+      run: git clang-format --diff $MERGE_BASE HEAD
```

Comment:
One active part of the script is missing though:
```
| grep -i -v '.mjs$' | grep -i -v LinkDef | grep -v -E '^D +' | sed -E 's,^.[[:space:]]+,,'
```
which filters:
* `.mjs` : javascript
* LinkDef : clang-format does an horrible job on `#pragma` lines
* '^D +' : avoid deleted files
* `| sed -E 's,^.[[:space:]]+,,'`: also related to delete files (5bc00921db08106e44196585d89615e99b7e7153)

Reply:
On a related note, as seen at https://github.com/root-project/root/actions/runs/15039480948/job/42267590977?pr=18713, the current filter for delete file might not be working well ... 

And similarly, it probably should not be looking at `.rootrc` files.

Reply:
> * `.mjs` : javascript

This doesn't pose a problem, since clang-format doesn't format this extension.
> * `LinkDef` : clang-format does an horrible job on `#pragma` lines

LinkDef is indeed checked, but here, maybe it would pay off to use `//@clang-format off`. This would have the benefit of not bothering you with suggestions or formatting when you run `git clang-format master` or similar on your machine.

> * `'^D +'` : avoid deleted files
> * `| sed -E 's,^.[[:space:]]+,,'`: also related to delete files ([5bc0092](https://github.com/root-project/root/commit/5bc00921db08106e44196585d89615e99b7e7153))

Deleted files are ignored by `git clang-format`, so this one also isn't a problem. I added a commit to my branch to show the result.

Reply:
I added one more commit to test `// clang-format off` for LinkDefs, and it works as expected. So maybe the best solution is to add this to all relevant LinkDefs, and then use the default `git clang-format`.

> And similarly, it probably should not be looking at .rootrc files.

These are also ignored by `git clang-format`.

Reply:
Then I wonder which part failed in https://github.com/root-project/root/actions/runs/15039480948/job/42267590977?pr=18713 ?

Reply:
> when you run git clang-format master or similar on your machine.

That would be a "disaster" :) ... at least until at clang-format 21.

>  // clang-format off for LinkDefs, and it works as expected. So maybe the best solution 

This is indeed the clearest ... but of course need a large-ish PR (of its own) to make the update.

Reply:
> This is indeed the clearest ... but of course needs a large-ish PR (of its own) to make the update.

This is now #18836

Reply:
@pcanal ready to approve?

Reply:
Mainly yes ... I would like to understood whether the failure seen at https://github.com/root-project/root/pull/18801#discussion_r2100295803 could affect this code (i.e. for that we need to understand/know what really failed there).

Reply:
> Mainly yes ... I would like to understood whether the failure seen at [#18801 (comment)](https://github.com/root-project/root/pull/18801#discussion_r2100295803) could affect this code (i.e. for that we need to understand/know what really failed there).

The failure in this PR is the script being too complicated:
If I pull Sergey's commit and get a `--name-status` from it, I get:
```
D<tab>roottest/root/io/abstractclass/.rootrc
M<tab>roottest/root/io/abstractclass/CMakeLists.txt
```
The script tries to suppress deleted files with `^D +` (space at least once or more). Therefore, we are explicitly invoking clang-format on a deleted file, and that will fail.

The good news is that all this is unnecessary with git-clang-format, since it automatically ignores deleted files.